### ISSUE TITLE
Invalid $number parameter: "" cannot be interpreted as a number

### ISSUE THREAD

**Issue by MonAssistantNumerique62** [PrestaShop/PrestaShop#23970]
Hello,

I encounter the following error on PS 1.7.7.2, PHP 7.3 in mutlistore context. The website runs on an OVH "performance" hosting.

"Invalid $number parameter: "" cannot be interpreted as a number

[PrestaShop\PrestaShop\Core\Localization\Exception\LocalizationException 0]"

I've read that closed bug report which describe the same problem : https://github.com/PrestaShop/PrestaShop/issues/22108

That error does not occurs when I'm on a single store, only when I'm on "all".

It occurs first when I've tried to ordered the articles by quantities.

I have only 1 currency (€) and 1 language (french).

I can't activate the PHP extensions as suggested in the previously mentioned bug report.

**Comment by florine2623** [comment]
Hello @MonAssistantNumerique62 ,

Please check your version of PHP : https://devdocs.prestashop.com/1.7/basics/installation/system-requirements/

You must follow the bug template so that we can test it : 
https://github.com/PrestaShop/PrestaShop/issues/new?assignees=&labels=&template=1_bug_report.md
![Capture d’écran 2021-04-09 à 16 48 06](https://user-images.githubusercontent.com/16019289/114198223-5e65d780-9953-11eb-8d36-c3ab53dbe69d.png)

**Comment by MonAssistantNumerique62** [comment]
Sorry... I'm wrong, its PHP 7.3 of course.

**Comment by MonAssistantNumerique62** [comment]
#### Describe the bug

I encounter the following error on PS 1.7.7.2, PHP 7.3 in mutlistore context. The website runs on an OVH "performance" hosting.

When trying to display the product list I receive that error :
"Invalid $number parameter: "" cannot be interpreted as a number

[PrestaShop\PrestaShop\Core\Localization\Exception\LocalizationException 0]"

I've read a closed bug report which describe the same problem : #22108

That error does not occurs when I'm on a single store, only when I'm on "all".

It occurs first when I've tried to ordered the articles by quantities.

I have only 1 currency (€) and 1 language (french).

I can't activate the PHP extensions as suggested in the previously mentioned bug report.

#### Expected behavior

Display of article list.

#### Steps to Reproduce

Steps to reproduce the behavior:

1. Go to 'all stores'
2. Click on 'Catalog->products'
3. Crash with message : "Invalid $number parameter: "" cannot be interpreted as a number

[PrestaShop\PrestaShop\Core\Localization\Exception\LocalizationException 0]"

**Screenshots**

<img width="1235" alt="image" src="https://user-images.githubusercontent.com/32057213/114217455-78f67b80-9968-11eb-8022-c27de2bbc57c.png">


#### Additional information

* PrestaShop version: 1.7.7.3
* PHP version: 7.3

**Comment by MonAssistantNumerique62** [comment]
To complete that bug report : I've restored a backup of my database. The product list is displayed successfully, but it crashed when ordering the list by available quantities. After that it crashed each time...

**Comment by florine2623** [comment]
The issue is fixed in PS 1.7.7.3 : https://github.com/PrestaShop/PrestaShop/issues/23376

**Comment by MonAssistantNumerique62** [comment]
@florine2623 No I've updated the site to PS 1.7.7.3 yesterday and still have that problem (I've done a cache cleaning after the upgrade).

It seems that this problem only appear on some articles. When filtering products by ID for some product it works, and for other I encounter that error. But I can't find what make the difference between a product that work and one that cause a crash...

**Comment by florine2623** [comment]
Hello @MonAssistantNumerique62 ,

I upgraded my store from PS 1.7.7.2 to PS 1.7.7.3, I can't reproduce your error.
https://drive.google.com/file/d/19xe7RQlqg6i5gr0DB83rESl8yRNKHSqu/view
I can filter my products by ID or quantity. 

We use GitHub issues only to discuss about bugs and new features in the PrestaShop project. If you have questions about using PrestaShop or third-party modules, or if you need help with your shop, please consider one of our [support plans](https://www.prestashop.com/support).

Alternatively, you can also ask for help in the [community forums](https://www.prestashop.com/forums/) or in the [public Slack channel](https://github.com/PrestaShop/open-source/tree/master/slack).

Thanks!

**Comment by MonAssistantNumerique62** [comment]
@florine2623 I've found the origin of that bug.

Here is a way to reproduce it :
1 - Need to be in multistore context
2 - Select "all shops" context
3 - Create an article
4 - Set it as "article with combination"
5 - Save the article
6 - Display the article list in backoffice

It will crash because the "final_price" field has a null value.

In my case the crash occurs because the combinations were created in the main store context instead of "all shops" context. So when I'm in "all shops" no combinations are displayed.

As a temporary solution I've added that code before line 391 in /src/Adapter/Product/AdminProductDataProvider.php :
if (!$product['price_final']) {
        $product['price_final']=$product['price'];
     }

Final price is incorrect but it avoid a null value and the subsequent crash.
